content/php-api-using-jwt/index.md

```diff
@@ -0,0 +1,226 @@
+
+Here is some sample code for the aforementioned process.
+```javascript
+const store = {};
+const loginButton = document.querySelector('#frmLogin');
+const btnGetResource = document.querySelector('#btnGetResource');
+const form = document.forms[0];
+
+// Inserts the jwt to the store object
+store.setJWT = function (data) {
+  this.JWT = data;
+};
+
+loginButton.addEventListener('submit', async (e) => {
+  e.preventDefault();
+
+  const res = await fetch('/authenticate.php', {
+    method: 'POST',
+    headers: {
+      'Content-type': 'application/x-www-form-urlencoded; charset=UTF-8'
+    },
+    body: JSON.stringify({
+      username: form.inputEmail.value,
+      password: form.inputPassword.value
+    })
+  });
+
+  if (res.status >= 200 && res.status <= 299) {
+    const jwt = await res.text();
+    store.setJWT(jwt);
+    frmLogin.style.display = 'none';
+    btnGetResource.style.display = 'block';
+  } else {
+    // Handle errors
+    console.log(res.status, res.statusText);
+  }
+});
+```
+We've already produced and submitted our JWT to the user. So now it's time to put the JWT to use on the user side.
+
+As previously stated, if the form submission is successful, we will display a button to obtain a timestamp. This button will invoke the GET method on our resource.php script. The resource.php script will then set the JWT obtained after successful authentication in the Authentication header.
+
+The following code will help you achieve this feat.
+
+```javascript
+btnGetResource.addEventListener('click', async (e) => {
+  const res = await fetch('/resource.php', {
+    headers: {
+      'Authorization': `Bearer ${store.JWT}`
+    }
+  });
+  const timeStamp = await res.text();
+  console.log(timeStamp);
+});
+```
+Run the program once you've written this code and enter your credentials into the form fields. A GET request will be sent when you click the submit button. Here is a sample GET request to assist you in making the right identification.
+```php
+GET /resource.php HTTP/1.1
+Host: yourhost.com
+Connection: keep-alive
+Accept: */*
+X-Requested-With: XMLHttpRequest
+Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE0MjU1ODg4MjEsImp0aSI6IjU0ZjhjMjU1NWQyMjMiLCJpc3MiOiJzcC1qd3Qtc2ltcGxlLXRlY25vbTFrMy5jOS5pbyIsIm5iZiI6MTQyNTU4ODgyMSwiZXhwIjoxNDI1NTkyNDIxLCJkYXRhIjp7InVzZXJJZCI6IjEiLCJ1c2VyTmFtZSI6ImFkbWluIn19.HVYBe9xvPD8qt0wh7rXI8bmRJsQavJ8Qs29yfVbY-A0
+```
+If you've followed along until here, everything should be working fine. Now let's begin with validating the JWT.
+
+To assist us with this operation, we've previously added the composer's autoloader function. We'll now use the preg match function to extract the token from the Bearer header. For this extraction, we'll use the preg match function and supply a regular expression.
+
+```php
+if (! preg_match('/Bearer\s(\S+)/', $_SERVER['HTTP_AUTHORIZATION'], $matches)) {
+    header('HTTP/1.0 400 Bad Request');
```

Comment:
Same recommendation as the previous code block comment

Reply:
Done, Please check